### PR TITLE
Merge of separate gyro project.

### PR DESCRIPTION
**Pull request by niklasborwell**

### CODE REVIEW

**Comment by davidhesselbom** [comment]
This gets a big thumbs up :+1: 

Unfortunately, it fails tests.

**Comment by niklasborwell** [comment]
I just realized I forgot to fix the tests....

**Comment by niklasborwell** [comment]
Ok, this is gonna take a while, but Im working on it.

**Comment by niklasborwell** [comment]
Ok, tests are working again.

File: source/math/FloatTransform3D.ooc [hunk 1]
```diff
@@ -206,10 +207,13 @@ FloatTransform3D: cover {
 		this o == other o &&
 		this p == other p
 	}
-	transformProjected: func (point: FloatPoint2D, focalLength: Float) -> FloatPoint2D {
+	transformProjected: func ~FloatPoint2D (point: FloatPoint2D, focalLength: Float) -> FloatPoint2D {
```
Comment (simonmika) [inline source/math/FloatTransform3D.ooc]: What does transformProjected do? It seams to transform it before it projects it? Is the name the right one?

Comment (niklasborwell) [inline source/math/FloatTransform3D.ooc]: Yes, this is a bad name. Maybe transformAndProject?

**Comment by niklasborwell** [comment]
Rebased on upstream.

**Comment by simonmika** [comment]
Can you rebase this as well?

**Comment by niklasborwell** [comment]
It is already rebased.

**Pull request merged**